src/cli.rs

```diff
@@ -182,8 +182,9 @@ impl Rover {
         // this only happens once a day automatically
         // we skip this check for the `rover update` commands, since they
         // do their own checks.
+        // we also skip this check for completion commands since they don't need it.
         // the check is also skipped if the `--skip-update-check` flag is passed.
-        if let Command::Update(_) = &self.command { /* skip check */
+        if let Command::Update(_) | Command::Completion(_) = &self.command { /* skip check */
```

Comment:
I don't think we want to exclude arbitrary commands from self-update checks. The only reason `Command::Update` is skipped is because it's _for_ updating.


```suggestion
        // the check is also skipped if the `--skip-update-check` flag is passed.
        if let Command::Update(_) = &self.command { /* skip check */
```